src/server.rs

```diff
@@ -15,113 +14,150 @@ use crate::configuration::Configuration;
 
+// TODO: passing configuration to Server
 pub struct Server {
     endpoint: String,
-    use_tls: bool
+    use_tls: bool,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 struct ErrorMessage {
-    message: String,
+    code: i32,
+    message: &'static str,
     details: Option<String>
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+struct Claims {
+    sub: String,
+    iss: String,
+    iat: i64,
+    exp: i64,
+}
+
 // TODO: move into implementation
-fn handler_vuejs<'a>(_: &mut Request, res: Response<'a>) -> MiddlewareResult<'a> {
+fn middleware_webui<'a>(_: &mut Request, res: Response<'a>) -> MiddlewareResult<'a> {
     let mut data = HashMap::<&str, &str>::new();
     data.insert("name", "Alex");
-    res.render("webui/dist/index.tpl", &data)
+    res.render("webui/dist/index.html", &data)
 }
 
-fn handler_logger<'a, D>(request: &mut Request<D>, response: Response<'a, D>) -> MiddlewareResult<'a, D> {
+fn middleware_logging<'a, D>(request: &mut Request<D>, response: Response<'a, D>) -> MiddlewareResult<'a, D> {
     crate::logger::print(&LogLevel::Information, format!("{} {}", request.origin.method, request.origin.uri).as_ref());
     response.next_middleware()
 }
 
+fn middleware_cors<'mw>(_req: &mut Request, mut res: Response<'mw>) -> MiddlewareResult<'mw> {
+    res.headers_mut().set_raw("Access-Control-Allow-Origin", vec![b"*".to_vec()]);
+    res.headers_mut().set_raw("Access-Control-Allow-Methods", vec![b"*".to_vec()]);
+    res.headers_mut().set_raw("Access-Control-Allow-Headers", vec![b"*".to_vec()]); //Origin, Authorization, X-Requested-With, Content-Type, Accept
+    res.next_middleware()
+}
+
 struct KvStoreMiddleware {
-    method: hyper::method::Method,
+    http_verb: hyper::method::Method,
     store: Arc<RwLock<KvStore>>,
 }
 
 impl<D> Middleware<D> for KvStoreMiddleware {
     fn invoke<'mw, 'conn>(&self, req: &mut Request<'mw, 'conn, D>, mut res: Response<'mw, D>) -> MiddlewareResult<'mw, D> {
-        // TODO: validate JWT token
-
         // Get the request body and retrieve the KV store
         let store = &*self.store.write().unwrap();
         let mut buffer = Vec::new();
         let body_size = req.origin.read_to_end(&mut buffer).unwrap();
 
         // Define some response headers
-        res.set(Server("Lucid 0.1.0".to_string()));
-
-        match self.method {
-            Method::Head => match req.param("key") {
-                Some(key) => match &store.get(key.to_string()) {
-                    Some(_) => {
-                        res.set(StatusCode::Ok);
-                        res.send("")
+        // TODO: use crate version
+        res.set(Server("Lucid 0.1.2".to_string()));
+
+        match req.origin.headers.get::<Authorization<Bearer>>() {
+            // TODO: get secret from configuration file
+            Some(header) => match decode::<Claims>(&header.token, "0cccdfddd23f9f740c9620a094daf1b260436059924cec32622f0e7ebc99bbe5".as_ref(), &Validation::default()) {
```

Comment:
What is this `0cccdfddd23f9f740c9620a094daf1b260436059924cec32622f0e7ebc99bbe5` string ?

Reply:
The private key